cds-snc/notification-admin
# Add support to display FR column headings in reports CSVs

# Summary | Résumé
This PR adds support for displaying French translated column headers in downloaded report CSVs. It also cleans up the headers for consistency, converting `email_address` and `phone_number` to `Email address` and `Phone number` for consistent case and styling in the headings.

![Screenshot 2023-10-03 at 10 17 54 AM](https://github.com/cds-snc/notification-admin/assets/7444334/43420e2e-caca-4ebd-bc6c-163238346be0)

![Screenshot 2023-10-03 at 10 18 45 AM](https://github.com/cds-snc/notification-admin/assets/7444334/7be20884-6a04-4f34-8819-27992cfcb770)

# Test instructions | Instructions pour tester la modification
## Test 1 - Basic sends
1. Create and send two bulk sends, one Email and one SMS
2. Navigate to the jobs page and switch the language to French
3. Download the report
4. Note that the column headings are in French and the case and spacing for email address and phone number are correct
5. Note that the column values for `Status` and `Type` are in French

## Test 2 - Custom columns
1. Create and send two bulk sends, one Email and one SMS both with custom headings
2. Navigate to the jobs page and switch the language to French
3. Download the report
4. Note that the column headings are in French and the Case and spacing for email address and phone number are correct
6. Note that custom column headings are preserved as they were input by the user

## Test Downloaded CSVs
1. Open both an SMS and Email CSV in MS Excel.
2. Note that accented characters are rendered properly
3. Check for any other encoding issues related to accented FR characters.
